plugins/rikki/loungestatistics/components/SlothStatistics.php

```diff
@@ -24,19 +24,49 @@ public function componentDetails()
     public $sloth = null;
     public $heroes = null;
     public $maps = null;
-    public $allseasons = null;
+    public $participatedSeasons = null;
+    public $selectedSeason = null;
 
     public function init()
     {
         $this->addJs('assets/js/datatables.min.js');
         $this->addJs('assets/js/loungestatistics.js');
         $this->addCss('assets/css/datatables.min.css');
+
         $this->sloth = Sloth::where('id', $this->property('sloth_id'))
                         ->with('gameParticipations', 'gameParticipations.team', 'gameParticipations.game', 'gameParticipations.game.map', 'gameParticipations.game.match', 'gameParticipations.hero', 'gameParticipations.game.teamOneFirstBan', 'gameParticipations.game.teamOneSecondBan', 'gameParticipations.game.teamTwoFirstBan', 'gameParticipations.game.teamTwoSecondBan')
                         ->first();
-        $this->allseasons = Season::where('reg_open', false)->get()->sortByDesc('created_at');
 
-        $this->calculateStats($this->allseasons->first());
+        $slothSeasonsSet = [];
+        foreach ($this->sloth->gameParticipations as $gP) {
+            if ($gP->hero == null || $gP->game == null || $gP->game->match == null) {
+                continue;
+            }
+            foreach ($gP->game->match->associatedSeasons() as $season) {
+                if ($season) {
+                    $slothSeasonsSet[$season->id] = $season;
+
+                    if ($season->is_active && ($this->selectedSeason == null || $season->created_at < $this->selectedSeason->created_at)) {
+                        $this->selectedSeason = $season;
+                    }
+                }
+            }
+        }
+
+        $this->participatedSeasons = array_values($slothSeasonsSet);
+        usort($this->participatedSeasons, function($a, $b) {
+            if ($a->created_at == $b->created_at) return 0;
+            return $a->created_at > $b->created_at ? 1 : -1;
+        });
+
+        if ($this->selectedSeason == null) {
+            $participatedSeasonsCount = count($this->participatedSeasons);
+            if ($participatedSeasonsCount > 0) {
+                $this->selectedSeason = $this->participatedSeasons[$participatedSeasonsCount - 1];
+            }
+        }
```

Comment:
This whole section probably works, but would look better code-wise by using Collections - have a look at map, sort, filter and unique.

Reply:
unique does not work here, but I used collections